llvm/test/CodeGen/DirectX/BufferLoad.ll

```diff
@@ -0,0 +1,102 @@
+; RUN: opt -S -dxil-op-lower %s | FileCheck %s
+
+target triple = "dxil-pc-shadermodel6.6-compute"
+
+declare void @scalar_user(float)
+declare void @vector_user(<4 x float>)
+
+define void @loadfloats() {
+  ; CHECK: [[BIND:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding
+  ; CHECK: [[HANDLE:%.*]] = call %dx.types.Handle @dx.op.annotateHandle(i32 217, %dx.types.Handle [[BIND]]
+  %buffer = call target("dx.TypedBuffer", <4 x float>, 0, 0, 0)
```

Comment:
Is it worth adding a test for a RWBuffer as well? I don't think it changes based on the buffer type, but thought I'd ask.

Reply:
I don't think it matters - the handle is just here because we need a valid type for the intrinsic to operate on and RWBuffer wouldn't make any difference.